code/addons/test/src/node/vitest-manager.ts

```diff
@@ -178,23 +178,63 @@ export class VitestManager {
     }
   }
 
-  private filterStories(
-    story: StoryIndexEntry | DocsIndexEntry,
-    moduleId: string,
-    tagsFilter: TagsFilter
+  private filterTestSpecifications(
```

Comment:
this is a refactor that combines the filtering logic previously within `runTests` with the now-removed `filterStories` function, so that the whole filtering logic is combined, and can be re-used across both `runTests` and `runAffectedTestsAfterChange`